RestaurentManagement/src/main/java/com/restapi/services/MenuService.java

```diff
@@ -27,67 +30,97 @@ public MenuService(MenuRepository menuRepository, TypeRepository typeRepository)
         this.typeRepository = typeRepository;
     }
 
-    public Menu createMenu(Menu menu) {
+    public MenuDTO createMenu(MenuDTO menu) {
         if (menu.getType().isEmpty()) {
             throw new RMValidateException(new ErrorDetail(
                     new Date().toString(),
-                    HttpStatus.BAD_REQUEST.getReasonPhrase(),
                     HttpStatus.BAD_REQUEST.value(),
+                    HttpStatus.BAD_REQUEST.getReasonPhrase(),
                     RMConstant.TYPE_BAD_REQUEST));
         }
-        List<Type> types = menu.getType();
-        List<Type> existingTypes = new ArrayList<>();
-        for (Type type : types) {
-            if (type.getId() != null && typeRepository.existsById(type.getId())) {
-                existingTypes.add(type);
-            } else {
-                Type newType = typeRepository.save(type);
-                existingTypes.add(newType);
-            }
+        List<Type> types = new ArrayList<>();
+        for (TypeDTO typeDTO : menu.getType()) {
+            Type type = new Type();
+            type.setType(typeDTO.getType());
+            types.add(type);
+        }
+        Menu menuToCreate = new Menu();
+        menuToCreate.setName(menu.getName());
+        menuToCreate.setDescription(menu.getDescription());
+        menuToCreate.setPrice(menu.getPrice());
+        menuToCreate.setImage(menu.getImage());
+        menuToCreate.setType(types);
+
+        typeRepository.saveAll(types);
```

Comment:
create a function in type service to save all types and create a menu will be call the save all type function